src/components/RoomNameInput.js

```diff
@@ -106,15 +114,25 @@ class RoomNameInput extends Component {
 
         let error = '';
 
-        // We error if the room name already exists. We don't care if it matches the original name provided in this
-        // component because then we are not changing the room's name.
-        if (isExistingRoomName && finalRoomName !== this.originalRoomName) {
-            error = this.props.translate('newRoomPage.roomAlreadyExistsError');
-        }
-
-        // Certain names are reserved for default rooms and should not be used for policy rooms.
-        if (_.contains(CONST.REPORT.RESERVED_ROOM_NAMES, finalRoomName)) {
-            error = this.props.translate('newRoomPage.roomNameReservedError');
+        // We show the error provided as props if we want to show error on demand rather than on every change
+        if (this.props.errorText && this.props.showErrorOnDemand) {
+            error = this.props.errorText;
+        } else {
+            // We error if the user doesn't enter a room name or left blank
+            if (finalRoomName === CONST.POLICY.ROOM_PREFIX) {
+                error = this.props.translate('newRoomPage.pleaseEnterRoomName');
+            }
+
+            // We error if the room name already exists. We don't care if it matches the original name provided in this
+            // component because then we are not changing the room's name.
+            if (isExistingRoomName && finalRoomName !== this.originalRoomName) {
+                error = this.props.translate('newRoomPage.roomAlreadyExistsError');
+            }
+
+            // Certain names are reserved for default rooms and should not be used for policy rooms.
+            if (_.contains(CONST.REPORT.RESERVED_ROOM_NAMES, finalRoomName)) {
+                error = this.props.translate('newRoomPage.roomNameReservedError');
+            }
```

Comment:
Here you just calculate the error as it was done before. You just need to add the new error cases from your changes.

Main change is in the JSX `errorText` prop of `TextInputWithPrefix`. 
it would become something like that.
```
errorText={shouldShowErrorOnchange ? this.state.error : this.props.errorText}
```

Reply:
nice